inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/BinlogReader.java

```diff
@@ -228,18 +240,24 @@ private Properties getEngineProps() {
         props.setProperty("table.whitelist", tableWhiteList);
         props.setProperty("database.whitelist", databaseWhiteList);
 
-        props.setProperty("offset.storage", FileOffsetBackingStore.class.getCanonicalName());
-        props.setProperty("offset.storage.file.filename", offsetStoreFileName);
         props.setProperty("offset.flush.interval.ms", offsetFlushIntervalMs);
-        props.setProperty("database.history", FileDatabaseHistory.class.getCanonicalName());
-        props.setProperty("database.history.file.filename", databaseStoreHistoryName);
         props.setProperty("database.snapshot.mode", snapshotMode);
         props.setProperty("database.history.store.only.monitored.tables.ddl", historyMonitorDdl);
         props.setProperty("database.allowPublicKeyRetrieval", "true");
         props.setProperty("key.converter.schemas.enable", "false");
         props.setProperty("value.converter.schemas.enable", "false");
         props.setProperty("include.schema.changes", includeSchemaChanges);
         props.setProperty("snapshot.mode", snapshotMode);
+        if ("schema_only_recovery".equals(snapshotMode)) {
+            props.setProperty("offset.storage", FlinkOffsetBackingStore.class.getCanonicalName());
+            props.setProperty(FlinkOffsetBackingStore.OFFSET_STATE_VALUE, serializeOffset());
+            props.setProperty("database.history", FlinkDatabaseHistory.class.getCanonicalName());
+        } else {
+            props.setProperty("offset.storage", FileOffsetBackingStore.class.getCanonicalName());
```

Comment:
no to to include FileDatabaseHistory and FileOffsetBackingStore here

Reply:
Create new class org.apache.inlong.agent.plugin.utils.InLongDatabaseHistory,  org.apache.inlong.agent.plugin.utils.InLongOffsetBackingStore